k8s/eks/bash/functions.sh

```diff
@@ -0,0 +1,312 @@
+  helm install testground-infra-redis --set auth.enabled=false --set master.nodeSelector='testground.node.role.infra: "true"' bitnami/redis
+} 
+
+# Until we find a better way for region var
+# tg_daemon_config_map(){
+#   kubectl apply -f ../kops/testground-daemon/config-map-env-toml.yml
+# } | tee -a ./log/$start-log/tg_daemon_config_map.log
+
+tg_daemon_config_map(){
+  kubectl create -f - <<EOF
+kind: ConfigMap
+apiVersion: v1
+metadata:
+  name: env-toml-cfg
+  namespace: default
+data:
+  .env.toml: |
+    ["aws"]
+    region = "$REGION"   
+
+    [runners."cluster:k8s"]
+    run_timeout_min             = 15
+    testplan_pod_cpu            = "100m"
+    testplan_pod_memory         = "100Mi"
+    collect_outputs_pod_cpu     = "1000m"
+    collect_outputs_pod_memory  = "1000Mi"
+    provider                    = "aws"
+    sysctls = [
+      "net.core.somaxconn=10000",
+    ]
+
+    [daemon]
+    listen = "0.0.0.0:8042"
+    slack_webhook_url = ""
+    github_repo_status_token = ""
+
+    [daemon.scheduler]
+    workers = 2
+    task_timeout_min  = 20
+    task_repo_type    = "disk"
+
+    [client]
+    endpoint = "localhost:8080"
+
+EOF
+}
+
+tg_daemon_service_account(){
+  kubectl apply -f ./yaml/tg-daemon-service-account.yml
+}
+
+tg_daemon_role_binding(){
+  kubectl apply -f ./yaml/tg-daemon-cluster-role.yml
+}
+
+tg_daemon_services(){
+  kubectl apply -f ./yaml/tg-testground-daemon-service.yml
+}
+
+tg_daemon_svc_sync_service(){
+  kubectl apply -f ./yaml/tg-sync-service.yml
+}
+
+tg_daemon_sync_service(){
+  kubectl apply -f ./yaml/tg-sync-service-deployment.yml
+}
+
+tg_daemon_sidecar(){
+  kubectl apply -f ./yaml/tg-ds-sidecar.yml
+}
+
+tg_daemon_deployment(){
+  kubectl apply -f ./yaml/tg-daemon-deployment.yml
+}
+
+log(){
+  tar czf ./log/$start-$CLUSTER_NAME-$CNI_COMBINATION.tar.gz /log/$start-log/
+  echo "##########################################"
+  echo "Log file generated with name $start-$CLUSTER_NAME-$CNI_COMBINATION.tar.gz"
+  echo "##########################################"
+  rm -rf ./log/$start-log/ 
+}
+
+# cluster_creation_manifest(){
+#   mkdir -p .cluster_manifest
+#   echo "$CLUSTER_NAME" >> clusters
+# }
+
+cleanup(){
+  mnt_target_id=$(aws efs describe-mount-targets --file-system-id $efs | jq -r ".MountTargets[] | .MountTargetId")
+  echo "Removing mount target with ID $mnt_target_id"
+  echo "The script will pause here for 20 seconds for the removal to complete"
+  aws efs delete-mount-target --mount-target-id $mnt_target_id
+  sleep 20
+  echo "Now removing EFS $efs"
+  aws efs delete-file-system --file-system $efs
+  echo "The script will pause here for additional 20 seconds"
+  sleep 20
+  echo "Now removing cluster, this may take some time"
+  eksctl delete cluster --name $CLUSTER_NAME
+  echo "Now removing EBS: $ebs"
+  aws ec2 delete-volume --volume-id $ebs
+  rm -f ./$CLUSTER_NAME.yaml
+  echo "Now fixing the .env file for the next usage"
+  head -n -3 .env >> .env.tmp && mv .env.tmp .env
```

Comment:
So when you say `between runs`, are you referring to cluster setup and then cluster deletion?
Because .env is used explicitly in these two cases, and nowhere else. So unless you edit it manually, you should have no issues with it.

Reply:
Yes, that's exactly what I mean.

Reply:
This logic has been updated, `.env` is no longer being written into.